pkg/scheduler/plugins/elasticquota/plugin_helper.go

```diff
@@ -48,17 +48,34 @@ func (g *Plugin) getPodAssociateQuotaName(pod *v1.Pod) string {
 	return quotaName
 }
 
-var GetQuotaName = func(pod *v1.Pod, quotaLister schedulinglisterv1alpha1.ElasticQuotaLister) string {
+func (g *Plugin) GetQuotaName(pod *v1.Pod) string {
 	quotaName := extension.GetQuotaName(pod)
 	if quotaName != "" {
 		return quotaName
 	}
-	eq, err := quotaLister.ElasticQuotas(pod.Namespace).Get(pod.Namespace)
+	eq, err := g.quotaLister.ElasticQuotas(pod.Namespace).Get(pod.Namespace)
 	if err == nil && eq != nil {
 		return eq.Name
 	} else if !errors.IsNotFound(err) {
 		klog.Errorf("Failed to Get ElasticQuota %s, err: %v", pod.Namespace, err)
 	}
+
+	eqList, err := g.quotaInformer.GetIndexer().ByIndex(extension.IndexQuotaNamespaces, pod.Namespace)
+	if err != nil {
+		return extension.DefaultQuotaName
+	}
+
+	for _, quota := range eqList {
+		eq, ok := quota.(*schedulerv1alpha1.ElasticQuota)
+		if !ok {
+			continue
+		}
+		namespaceSet := sets.NewString(extension.GetAnnotationQuotaNamespaces(eq)...)
+		if namespaceSet.Has(pod.Namespace) {
```

Comment:
If ElasticQuotas is indexed via the indexer `annotation.namespaces`, is there a need to detect namespaces again?